.git_archival.txt

```diff
@@ -0,0 +1,4 @@
+node: $Format:%H$
+node-date: $Format:%cI$
+describe-name: $Format:%(describe:tags=true,match=*[0-9]*)$
+ref-names: $Format:%D$
```

Comment:
can you explain why we need this file? Looking at hatch-vcs, I don't see any requirement of this configuration.

Reply:
This file allows users to install `Zarr` from Git archives without encountering version issues. This is not a `hatch-vcs` thing, instead this file should've been introduced when `setuptools-scm` was introduced in Zarr - https://setuptools-scm.readthedocs.io/en/latest/usage/#builtin-mechanisms-for-obtaining-version-numbers. `hatch-vcs` uses `setuptools_scm` underneath.

Scientific Python also [recommends adding git archival support](https://learn.scientific-python.org/development/guides/packaging-simple/) -
```
You should also add these two files:

.git_archival.txt:

node: $Format:%H$
node-date: $Format:%cI$
describe-name: $Format:%(describe:tags=true,match=*[0-9]*)$
ref-names: $Format:%D$

And .gitattributes (or add this line if you are already using this file):

.git_archival.txt  export-subst

This will allow git archives (including the ones generated from GitHub) to also support versioning.
```

Reply:
Makes sense. Thanks for the explanation.